File: Frontend/src/app/viewModels/Types/ErrorType.ts
/**
 * Type class that represents an error with a name and description
 */
export class ErrorType {

    /**
     * Construct's an instance of {@link ErrorType}
     * 
     * @param name error's name
     * @param description error's description
     */
    constructor(
        public name: string,
        public description: string
    ) { }
}

File: Backend/Data/ViewModels/Types/ErrorType.cs
namespace Backend.Data.ViewModels.Types
{
    /// <summary>
    /// Class that defines comment type
    /// </summary>
    public class ErrorType
    {
        public string Name { get; set; }
        public string Description { get; set; }

        /// <summary>
        /// Class' constructor method 
        /// </summary>
        /// <param name="name">Name of the variable that is wrong</param>
        /// <param name="description">Error descritpion</param>
        public ErrorType(string name, string description)
        {
            this.Name = name;
            this.Description = description;
        }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        public ErrorType() {
            this.Name = null!;
            this.Description = null!;
        }
    }
}
